.github/workflows/performance-benchmark.yml

```diff
@@ -0,0 +1,66 @@
+name: github-benchmarks
+on:
+  push:
+    branches:
+      - develop
+      - performance-results-visualization
+
+jobs:
+  CI:
+    continue-on-error: true
+    strategy:
+      matrix:
```

Comment:
Results on GitHub hosted runners (free tier) are wildly unstable. We might as well remove the `matrix` and limit ourselves to single configuration. Leaving at least `g++` and `clang` variants for now.